.circleci/config.yml

```diff
@@ -59,31 +59,41 @@ jobs:
       - run:
           name: Verify lockfile
           command: cargo update -w --locked
-      - run:
-          name: Build
-          command: cargo build --release
-      - run:
-          name: Clippy check
-          command: ./scripts/clippy_check.sh
-      - run:
-          name: Format check
-          command: ./scripts/format_check.sh
-      - run:
-          name: Code spell check
-          command: ./scripts/check_spelling.sh
+      # - run:
+      #     name: Build
+      #     command: cargo build --release
+      # - run:
+      #     name: Clippy check
+      #     command: ./scripts/clippy_check.sh
+      # - run:
+      #     name: Format check
+      #     command: ./scripts/format_check.sh
+      # - run:
+      #     name: Code spell check
+      #     command: ./scripts/check_spelling.sh
       - run:
           name: Save cache
           command: .circleci/scripts/cache.sh save
       - run:
           name: Cleanup old cache
           command: .circleci/scripts/cache.sh cleanup
-      - run:
-          name: Unit tests
-          command: cargo test --lib --bins --no-fail-fast
-      - run:
-          name: Integration tests
-          # fails if jobs not limited; read more in https://github.com/0xSpaceShard/starknet-devnet-rs/issues/378
-          command: cargo test --jobs 9 --test '*' --no-fail-fast
+      # - run:
+      #     name: Unit tests
+      #     command: cargo test --lib --bins --no-fail-fast
+      # - run:
+      #    name: Integration tests
+      #    # fails if jobs not limited; read more in https://github.com/0xSpaceShard/starknet-devnet-rs/issues/378
+      #    command: cargo test --jobs 9 --test '*' --no-fail-fast
+      - when:
+          condition:
+            not:
+              equal: [ main, << pipeline.git.branch >> ]
```

Comment:
Doesn't this add a lot of time to the workflow? Locally for me a single publish dry run takes several minutes. Wouldn't it make more sense to actually only run it on `main`?

Reply:
on main there will be dry run an publish. The idea is to not have changes into main that cannot be published to crates.io.

Reply:
Also will increase the time